File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/model/database/MTGDatabaseHelper.java
package com.dbottillo.mtgsearchfree.model.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import com.dbottillo.mtgsearchfree.BuildConfig;
import com.dbottillo.mtgsearchfree.sqliteasset.SQLiteAssetHelper;

/**
 * Helper for access the card database, only on read mode
 * the database is created from {@link CreateDatabaseHelper}
 * and then copied to database folder from the library {@link SQLiteAssetHelper}
 */
public class MTGDatabaseHelper extends SQLiteAssetHelper {

    private static final String DATABASE_NAME = "mtgsearch.db";

    public MTGDatabaseHelper(Context context, String name, SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    public MTGDatabaseHelper(Context context) {
        this(context, DATABASE_NAME, null, BuildConfig.DATABASE_VERSION);
        setForcedUpgrade();
    }

}

File: core/src/main/kotlin/com/dbottillo/mtgsearchfree/model/storage/database/MTGDatabaseHelper.kt
package com.dbottillo.mtgsearchfree.model.database

import android.content.Context

import com.dbottillo.mtgsearchfree.core.BuildConfig
import com.dbottillo.mtgsearchfree.sqliteasset.SQLiteAssetHelper

/**
 * Helper for access the card database, only on read mode
 * the database is created from [CreateDatabaseHelper]
 * and then copied to database folder from the library [SQLiteAssetHelper]
 */
class MTGDatabaseHelper(context: Context) : SQLiteAssetHelper(context, "mtgsearch.db", BuildConfig.DATABASE_VERSION)
